enterprise/server/backends/userdb/userdb.go

```diff
@@ -931,12 +915,60 @@ func (d *UserDB) GetImpersonatedUser(ctx context.Context) (*tables.User, error)
 	return user, err
 }
 
-func (d *UserDB) DeleteUser(ctx context.Context, userID string) error {
-	// Permission check.
-	_, err := d.GetUserByID(ctx, userID)
+type userGroupRole struct {
+	UserUserID string
+	tables.GroupRole
+}
+
+func (d *UserDB) GetUsersBySubIDPrefixWithoutAuthCheck(ctx context.Context, subIDPrefix string) ([]*tables.User, error) {
+	rq := d.h.NewQuery(ctx, "userdb_get_users_by_sub_id_prefix").Raw(`
+			SELECT * FROM "Users" WHERE sub_id LIKE ?
+	`, subIDPrefix+"%")
+	users, err := db.ScanAll(rq, &tables.User{})
 	if err != nil {
-		return err
+		return nil, err
 	}
+
+	byID := map[string]*tables.User{}
+	for _, u := range users {
+		byID[u.UserID] = u
+	}
+
+	for len(users) > 0 {
+		count := min(len(users), 200)
+		userIDs := []string{}
+		for _, u := range users[:count] {
+			userIDs = append(userIDs, u.UserID)
+		}
+		rq := d.h.NewQuery(ctx, "userdb_get_users_by_sub_id_prefix").Raw(`
```

Comment:
nit: do you want to use a different query name for this one?

```suggestion
		rq := d.h.NewQuery(ctx, "userdb_get_user_groups_by_sub_id_prefix").Raw(`
```